src/rdkafka_ssl.c

```diff
@@ -1298,6 +1352,47 @@ int rd_kafka_ssl_ctx_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
         }
 #endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+        if (rk->rk_conf.ssl.engine_location && !rk->rk_conf.ssl.engine) {
+                char *engine_errstr = NULL;
+
+                /* Openssl loads an engine as dynamic id and stores it in internal
+                 * list, as per LIST_ADD command below. If engine already exists
+                 * in internal list, it is supposed to be fetched using engine id. 
+                 */
+                rk->rk_conf.ssl.engine = ENGINE_by_id(rk->rk_conf.ssl.engine_id);
+                if (!rk->rk_conf.ssl.engine) {
+                        rk->rk_conf.ssl.engine = ENGINE_by_id("dynamic");
```

Comment:
Not needed if "dynamic" is set as default in the conf_t struct.

Reply:
as far as i understand from my openssl testing, flow is first we need to use "dynamic" first time for any engine. subsequent loads will require "engine id"